sccameraview/src/main/java/com/example/sccameraview/BaseCameraView.java

```diff
@@ -53,20 +60,24 @@ public void setCalculatedHeight(int calculatedHeight) {
     }
 
     CamcorderProfile getCamcorderProfile() {
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_HIGH)) {
+            return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_HIGH);
+        }
         if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_480P)) {
             return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_480P);
         }
-
         return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_LOW);
     }
 
     abstract void openCamera();
 
+    public abstract void switchCamera();
```

Comment:
camera is opened based on frontFacingCameraActive attribute, which is set true in the constructor, is that a bad practice?